ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -282,6 +282,12 @@ static int pdo_sqlite_get_attribute(pdo_dbh_t *dbh, zend_long attr, zval *return
 	return 1;
 }
 
+static bool pdo_sqlite_in_transaction(pdo_dbh_t *dbh)
+{
+	pdo_sqlite_db_handle* H = (pdo_sqlite_db_handle*) dbh->driver_data;
+	return H->db && sqlite3_get_autocommit(H->db) == 0;
+}
+
```

Comment:
- SQLite3 does not allow you to explicitly turn off autocommit mode other than manually starting a transaction
- Manual transactions always exist when autocommit mode is off

Since this kind of behavior is specific to SQLite, it might be better to leave a comment.